engine/lib/membrane_rtc_engine/engine.ex

```diff
@@ -715,39 +737,93 @@ defmodule Membrane.RTC.Engine do
          _ctx,
          state
        ) do
-    if Map.has_key?(state.endpoints, endpoint_id) do
-      endpoint = Map.get(state.endpoints, endpoint_id)
-      track = Endpoint.get_track_by_id(endpoint, track_id)
-
-      if track != nil and track.metadata != track_metadata do
-        endpoint = Endpoint.update_track_metadata(endpoint, track_id, track_metadata)
-        state = put_in(state, [:endpoints, endpoint_id], endpoint)
-
-        actions =
-          state.subscriptions
-          |> Map.values()
-          |> Enum.flat_map(&Map.values/1)
-          |> then(&(&1 ++ state.pending_subscriptions))
-          |> Enum.filter(&(&1.track_id == track_id))
-          |> Enum.map(& &1.endpoint_id)
-          |> Enum.map(
-            &{:notify_child,
-             {{:endpoint, &1},
-              {:track_metadata_updated, Endpoint.get_track_by_id(endpoint, track_id)}}}
-          )
-
-        dispatch(%Message.TrackMetadataUpdated{
-          endpoint_id: endpoint_id,
-          track_id: track_id,
-          track_metadata: track_metadata
-        })
+    with {:ok, endpoint} <- Map.fetch(state.endpoints, endpoint_id),
+         track when track != nil <- Endpoint.get_track_by_id(endpoint, track_id),
+         true <- track.metadata != track_metadata do
+      endpoint = Endpoint.update_track_metadata(endpoint, track_id, track_metadata)
+      state = put_in(state, [:endpoints, endpoint_id], endpoint)
 
-        {actions, state}
-      else
+      actions =
+        prepare_track_notifications(
+          state,
+          track_id,
+          {:track_metadata_updated, Endpoint.get_track_by_id(endpoint, track_id)}
+        )
+
+      dispatch(%Message.TrackMetadataUpdated{
+        endpoint_id: endpoint_id,
+        track_id: track_id,
+        track_metadata: track_metadata
+      })
+
+      {actions, state}
+    else
+      _other ->
         {[], state}
-      end
+    end
+  end
+
+  defp handle_endpoint_notification(
+         {:enable_track_variant, track_id, encoding},
```

Comment:
Nitpick: 
```suggestion
         {:enable_track_variant, track_id, variant},
```

Could we check other callbacks too?